### ISSUE TITLE
Test fails on Debian unstable

### ISSUE THREAD

**Issue by rockdaboot** [getdnsapi/getdns#300]
On Debian unstable:

After fresh git clone, git submodule update --init, libtoolize -ci, autoreconf -fi, ./configure, make, make test.
```
...
check_getdns: ./context.c:1579: getdns_context_destroy: Assertion `context->processing == 0' failed.
check_getdns: ./context.c:1579: getdns_context_destroy: Assertion `context->processing == 0' failed.
check_getdns: ./context.c:1579: getdns_context_destroy: Assertion `context->processing == 0' failed.
...
99%: Checks: 239, Failures: 1, Errors: 0
./check_getdns_context_set_dns_transport.h:166:F:Positive:getdns_context_set_dns_transport_stub_5:0: Packet not trucated as expected: expecting 1: 1, but received: 0: Good
if test 0 = 1 ; then CK_TIMEOUT_MULTIPLIER=2 CK_LOG_FILE_NAME=".log" ./ || echo " failed" >> fails; fi
if test 0 = 1 ; then CK_TIMEOUT_MULTIPLIER=2 CK_LOG_FILE_NAME=".log" ./ || echo " failed" >> fails; fi
if test 0 = 1 ; then CK_TIMEOUT_MULTIPLIER=2 CK_LOG_FILE_NAME=".log" ./ || echo " failed" >> fails; fi
test ! -e fails
Makefile:152: recipe for target 'test' failed
```

My `make` is an alias to `make -j4`
Commit is 80897d2b8644a5a15958b50cebfae4362df7d205
gcc (Debian 6.3.0-14) 6.3.0 20170415

**Comment by wtoorop** [comment]
Thank you Tim,

I am aware of this issue.  It is caused by dnsmasq not handling truncation correctly.  I have to file a bug report with them, but I'll keep this ticket open as an reminder that I should do that.

besides this...
  * The assertion messages look scary but are intentional (testing certain error cases)
  * Currently getdns doesn't support simultaneous jobs (jet), see also issue: #166

**Comment by KonomiKitten** [comment]
Is there anyway to skip tests so building is possible? I don't see an option in configure for it.

**Comment by wtoorop** [comment]
Ah... you want to skip this specific test?

**Comment by KonomiKitten** [comment]
> Ah... you want to skip this specific test?

That would be handy I am currently skipping all tests to build.

**Comment by wtoorop** [comment]
Konomi, I've made it so that the critical test will be skipped when the SKIP_DNSMASQ_ISSUE environment variable is set.  This will be in the 1.3.0 release, which I will do tomorrow.

**Comment by KonomiKitten** [comment]
@wtoorop thank you very much for your time and work :)

**Comment by wtoorop** [comment]
You're very welcome :)
Actually, I've just managed to improve this and detect dnsmasq presence and skip that specific unit test automatically, so no need to set an environment variable either.
This effectively resolves this issue.
The fix will be in the 1.3.0 release tomorrow.
